src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs

```diff
@@ -72,5 +72,31 @@ public override string ToString()
         {
             return !(left == right);
         }
+
+        // this ugly method exists to not break backward compatibility
+        // for cases where users use the old `OSX` property on macOS
+        private bool AreOSXAndMacOS(string left, string right)
```

Comment:
I think I'd prefer if we did the canonicalization for OSX/macOS in the constructor, so that Equals can stay simple.

So e.g.:

```csharp
private OSPlatform(string osPlatform)
{
    ...
    if (osPlatform.Equals("OSX", StringComparison.OrdinalIgnoreCase)) osPlatform = "MACOS";

    _osPlatform = osPlatform;
}
```